packages/realm/binding/jsi/react_scheduler.cpp

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "react_scheduler.h"
+
+#include <realm/object-store/util/scheduler.hpp>
+
+#include <ReactCommon/CallInvoker.h>
+#include <ReactCommon/SchedulerPriority.h>
+
+#include <thread>
+
+namespace realm::js::react_scheduler {
+
+using Scheduler = realm::util::Scheduler;
+
+// std::shared_ptr<facebook::react::CallInvoker> js_call_invoker_{};
+std::shared_ptr<Scheduler> scheduler_{};
+
+
+class ReactScheduler : public realm::util::Scheduler {
```

Comment:
This is very minor, but this type and the global variable should be in an anonymous namespace so that the symbols are not exported.